### ISSUE TITLE
Feature request: Multiple filters and URL w. filters

### ISSUE THREAD

**Issue by specimen151** [ServiceStack/Studio#2]
Feature request/feedback:
(don't think this suitable for SO, feel free to delete this)

In the AutoQuery UI, we were able to have more than one filter. Not it seems there is only one filter.

In AutoQuery UI, the resulting URL would be displayed, allowing for easy experimentation within the UI, then copy that URL to code/scripts.

Finally, will it eventually work with custom services (not only AutoQuery)? Of course it won't know about the filters in the same way, so the functionality would be limited. Or is that out of scope?

**Comment by mythz** [comment]
Hi, please post any feature requests on UserVoice: [servicestack.uservoice.com](https://servicestack.uservoice.com/forums/176786-feature-requests) so they can be recorded, demand for it measured and prioritized.

> In the AutoQuery UI, we were able to have more than one filter. Not it seems there is only one filter.

It only supports 1 major search condition but each column can have an additional filter as seen in:

![image](https://user-images.githubusercontent.com/89361/130796785-42cd9e23-c512-4d89-842a-8e0269e1c299.png)

Refer to the docs for the [Search Filters syntax](https://docs.servicestack.net/studio-autoquery#search-filters) that can be used in column filters. It's unlikely we'll make the Search UI any more complicated than this.

> In AutoQuery UI, the resulting URL would be displayed, allowing for easy experimentation within the UI, then copy that URL to code/scripts.

The `Output` window does let you see a log of all requests the UI makes which does have a copy URL icon, unfortunately the Desktop App UI itself isn't making the API queries directly, it crafts a special request to the backend which makes the API Request on its behalf  (i.e. to workaround CORS) which also adds any necessary auth info to the proxied request. 

But I've just added a **Copy URL** icon that simulates creating the URL on the client which you should now be able to use to make requests in your normal browser so if you restart Studio it should open the latest UI with the latest changes.

> Finally, will it eventually work with custom services (not only AutoQuery)? Of course it won't know about the filters in the same way, so the functionality would be limited. Or is that out of scope?

Not this specific AutoQuery UI which is designed for AutoQuery Services but we're exploring creating a generic API tool that can make requests for any API.

**Comment by specimen151** [comment]
My apologies, didn't know about UserVoice. Good idea. The "Copy URL" is there now, thank you for that.

**Just FYI**, the column filters aren't visible (no filter icon, no pencils), which I guess is because of the super simple configuration (just AutoQuery - no AutoCRUD). 

```Plugins.Add(new AutoQueryFeature { MaxLimit = 100});```

Which is directly from `x mix autoquery`. 

![image](https://user-images.githubusercontent.com/3056108/130812839-2556aef3-4817-4066-9cb8-2179934c5a9b.png)

**Comment by mythz** [comment]
It wasn't available because it couldn't find any AutoQuery CRUD operations but it that wasn't necessary as it just needs query access, so if you restart Studio it should now appear.

**Comment by specimen151** [comment]
I'm sorry to report that it still does not appear. I tested on two different projects, v.5.12.1 of all dll's. I have one DTO with AutoCRUD DTOs (e.g. `public class CreateTemplate : ICreateDb<Template>, IReturn<CreateTemplateResponse>`),  did get the filter column (and more), the others with only.

This one did not get filter:
```
   public partial class QueryDepartments
        : QueryDb<Department>, IReturn<QueryResponse<Department>>, IGet
    {
        public virtual int? DepNo { get; set; }
    }
```

Regarding running Studio, I did `-clean`, `uninstall`, updated "app" tool, etc. Just to make sure I had the newest Studio.
